flink-mesos/src/test/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorkerTest.java

```diff
@@ -52,4 +52,19 @@ public void canGetPortKeys() {
 		assertEquals("port key must be correct", "anotherport", iterator.next());
 	}
 
+	@Test
+	public void canGetNoPortKeys() {
+		// Setup
+		Configuration config = new Configuration();
+
+		// Act
+		Set<String> portKeys = LaunchableMesosWorker.extractPortKeys(config);
+
+		// Assert
+		assertEquals("Must get right number of port keys", 2, portKeys.size());
+		Iterator<String> iterator = portKeys.iterator();
+		assertEquals("port key must be correct", LaunchableMesosWorker.TM_PORT_KEYS[0], iterator.next());
```

Comment:
I find this test, and the implementation of `extractPortKeys` not ideal.

1. From the signature of `extractPortKeys` it is not obvious that the elements will be ordered.
1. Generally speaking there is no order imposed on the elements of a `Set` but the order of iteration should be deterministic for all collections [1], i.e., if you iterate twice over the same instance of the collection, you get the same order, which should be good enough for your use case.
1. Considering all above, the assertions can be simplified as 
```
assertThat(portKeys, containsInAnyOrder(LaunchableMesosWorker.TM_PORT_KEYS[0], LaunchableMesosWorker.TM_PORT_KEYS[1]));
```
Note that this also asserts that the size matches.

[1] https://stackoverflow.com/a/27210701/9203198